src/modules/git_state.rs

```diff
@@ -0,0 +1,165 @@
+use ansi_term::Color;
+use git2::{Repository, RepositoryState};
+use std::path::Path;
+
+use super::{Context, Module};
+
+/// Creates a module with the state of the git repository at the current directory
+///
+/// During a git operation it will show: REBASING, BISECTING, MERGING, etc.
+/// If the progress information is available (e.g. rebasing 3/10), it will show that too.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let repo_root = context.repo_root.as_ref()?;
+    let mut repository = Repository::open(repo_root).ok()?;
+    let state_description = get_state_description(&mut repository);
+
+    if let StateDescription::Clean = state_description {
+        return None;
+    }
+
+    let mut module = context.new_module("git_state")?;
```

Comment:
Can we move this to the top of the function? A la #283, touching the filesystem is sometimes a very expensive operation, and we'd like to avoid doing it at all if the module is disabled.

Reply:
Ah, good point. I actually missed that `new_module` checked the `disabled` setting. 👍 

I'll fix that shortly.